### PR TITLE
[SCHEMA] Create macro to render fields within objects

### LINKED ISSUE #841: `Genetics` subobject in the schema
The `genetics` object in dataset_description has 3 sub-objects: 
- `Dataset` 
- `Database`
- `Descriptors`

https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/08-genetic-descriptor.html#dataset-description

Currently the schema has a single yaml file for each sub-object but there is no yaml file for the `Genetics`.
Moreover the nested objects is not captured by the schema.
And the way the sub-objects are mentioned in the rendred table involve a dot notation `Genetics.Dataset` that appears nowhere else in the spec.

### Suggestion

- create yaml file for `Genetics`
- rewrite the part of the the Genetics to remove the dot notation from the table

### Possible issues

Not sure how to formalize the nested structure in the schema itself:

Should the sub-objects of `Genetics` be defined in the yaml of `Genetics`?

Should we keep the yaml files for each sub-object, mention in the README that this dot notation means they are sub-objects?

I don't think it makes sense to have pieces of metadata just called `Dataset`, `Database`,  `Descriptors` when they only relate to `Genetics`.

Tagging @tsalo and @CPernet for ideas and feedback on this.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #841.

Basically, we can move sub-object definitions into the actual object definitions now that we have a function to render sub-objects directly.

Here's what the new Genetics metadata sub-object table looks like:

image

I can adopt any kind of notation for the key name column. Maybe `Genetics[Dataset]`? `[Genetics]Dataset`?

### CODE REVIEW

**Review by sappelhoff — APPROVED** [review]
lgtm, +1 to merge after a rebase and observing the "review time"

File: src/04-modality-specific-files/08-genetic-descriptor.md [hunk 1]
```diff
@@ -27,18 +27,19 @@ Datasets linked to a genetic database entry include the following REQUIRED or OP
...
+{{ MACROS___make_subobject_table(
+   ("objects", "metadata", "Genetics"),
    {
-      "Genetics.Dataset": "REQUIRED",
-      "Genetics.Database": "OPTIONAL",
-      "Genetics.Descriptors": "OPTIONAL",
-   }
+      "Dataset": "REQUIRED",
+      "Database": "REQUIRED",
+      "Descriptors": "REQUIRED",
```
Comment (Remi-Gau) [inline src/04-modality-specific-files/08-genetic-descriptor.md:41]: Did those get upgrade to REQUIRED?

Comment (Remi-Gau) [reply src/04-modality-specific-files/08-genetic-descriptor.md]: hahaha I am not completely useless it seems

Comment (tsalo) [reply src/04-modality-specific-files/08-genetic-descriptor.md]: 😆

**Comment by tsalo** [comment]
If no one raises an issue with this one I'll merge it tomorrow.

**Pull request merged**